Funding bylaw (#1295)
php-fig/fig-standards: bylaws/009-funding.md

# Funding

## Scope and objectives

> PHP-FIG is, as it often happens in Open Source communities, an organization composed by volunteers only; this doesn't exclude the fact that the organization itself may have some costs to sustain itself, like for maintaining a couple of tools such as a site and an email account. For such and similar reasons, we need a way to raise and govern money inside our organization.

Reviewer: ```suggestion
PHP-FIG is, as it often happens in Open Source communities, an organization composed by volunteers only; this doesn't exclude the fact that the organization itself may have some costs to sustain itself, for example maintaining tools such as a site and an email account. For such and similar reasons, we need a way to raise and govern money inside our organization.
```

This document defines the roles, rules and processes that must be followed inside the PHP-FIG in all matters regarding money, to guarantee fairness, transparency and impartiality. This document intentionally describes the only allowed processes and ways to raise, administer and spend money, so that any change to such processes or any variation in budget allocation requires a bylaw change vote. 

> ## Fiscal hosting

Reviewer: ```suggestion
## Fiscal hosting

```

I find it much easier to read markdown files if there's an extra blank line between the titles and body.  (Same elsewhere in the file.)

For handling money and fiscal matters, the PHP-FIG will use [OpenCollective](https://opencollective.com/) as a fiscal host.

> As for any other tool pertaining to the PHP-FIG, secretary will be the only ones to have full administrative access to PHP-FIG's OpenCollective account, but they will not have any decision power over it, and they are instead bounded by this document.

Reviewer: ```suggestion
All three Secretaries, but only the Secretaries, will have full administrative access to PHP-FIG's OpenCollective account.  They do not have discretionary authority, however, and are bound to act as directed by this bylaw.
```

> All monetary contributions will flow through it, so that any tax matter will be handled automatically, and every financial decision will be published and publicly available through their budgetary tools.

Reviewer: ```suggestion
All monetary contributions will flow through OpenCollective, so that any tax matter will be handled automatically, and every financial decision will be published and publicly available through OpenCollective's budgetary tools.
```

> ## Raising money
> ### Through OpenCollective
> Contributions will be accepted through OpenCollective; PHP-FIG members will refrain from suggesting or requiring contributions from other members or the community at large if the monetary needs of the PHP-FIG are already met (see "Spending money" section).
> 
> ### Through TideLift
> As the main way of raising money, PHP-FIG will use [TideLift](https://tidelift.com/).
> The TideLift account will pay out any dividend to the OpenCollective account.

Reviewer: ```suggestion
## Raising money

PHP-FIG accepts donations through multiple channels.  Only the Secretaries may actively solicit donations from any party.  No other PHP-FIG member may do so.

### Through OpenCollective

Direct contributions will be accepted through OpenCollective;.

### Through TideLift

Contributions will be accepted through [TideLift](https://tidelift.com/).  They will pay out to the OpenCollective account.

### Through Thanks.dev

Contributions will be accepted through [Thanks.dev](https://thanks.dev).  They will pay out to the OpenCollective account.
```

This puts the fundraising responsibility (such as it is) exclusively on the Secretaries, but gives us easy "pluggable" flexibility to add/remove supported channels here over time.

> The TideLift account will pay out any dividend to the OpenCollective account.

Author: Also, for the record, TideLift requires some duties to be performed: https://support.tidelift.com/hc/en-us/articles/4406288074260-Lifter-tasks-overview

We basically do already all of those, apart from the security/vulnerability stuff, which would be good to do anyway.

## Spending money

> Any payment has to be budgeted and described in this document. Any payment has to be justified and aligned with the mission of the PHP-FIG, which is to push forward the evolution of the PHP community, through the definition of common standard and tools, and by helping PHP Open Source maintainers which follow the same goals.

Reviewer: ```suggestion

Funds donated to the PHP-FIG are to be used for non-personnel operating expenses.  PHP-FIG does not pay individual contributors to FIG standards or working groups.

All operating or capital expenses must be approved by an Approval Vote of the Core Committee.  The expense must include whether it is one time, monthly, or annually, and if it is ongoing or one-time.  All expenses must be justified by their contribution to the mission of PHP-FIG.

Expenses approved by the Core Committee will be listed below.
```

> Any payment has to be budgeted and described in this document. Any payment has to be justified and aligned with the mission of the PHP-FIG, which is to push forward the evolution of the PHP community, through the definition of common standard and tools, and by helping PHP Open Source maintainers which follow the same goals.

Reviewer: This also allows the CC to pre-approve spending 5 years worth of domain registration, or whatever.

### Overfunding, and giving back to the community

> PHP-FIG is allowed to store in its OpenCollective account enough funds to sustain itself for three years in the future, according to the current approved budget.
> At least once a year, any exceeding funds will be forwarded to the [PHP Foundation](https://opencollective.com/phpfoundation), as a way to give back to the community, and further increase its growth.

Reviewer: ```suggestion

PHP-FIG will maintain financial reserves to account for three years of the currently approved budget.

At least once a year, any surplus funds will be forwarded to the [PHP Foundation](https://opencollective.com/phpfoundation) as a way to further support the PHP ecosystem.
```

### Approved expenses

> This section will enumerate any approved expense.

Reviewer: I am skeptical of making the budget part of the bylaw.  That means any new budgetary change requires a full vote of all project reps as well as the CC, and... well, projects are not exactly active.  I'd prefer to have the budget by specifically as specified by the CC, not a full bylaw change.

We can wordsmith that if you're OK with that approach.

Author: Done in de5bbe8

#### Domain, site, email
The main recurring expense will be:
 * the domains (`php-fig.com` and `php-fig.org`)
 * the email account (`info@php-fig.org`)

Currently, those are and will be bought using NameCheap as a registrar.

> If paying for multiple years at once allows for a discount, and if enough funding is available, this should be done with an upper limit of five years in the future.

Reviewer: I think this section can be combined with the budget listed below, as a proposed initial budget.  Approving the bylaw would include approving the initial budget requests.

### Chronological list of budgetary decisions
This section will contain a chronological list of budgeting decisions, either recurring or not.

| Since | Expense | Approximate amount to be spent |
|-------|---------|--------------------------------|
| TBA   | Domains | 2 x ~15 USD / year             |

> | TBA   | Email   | ~15 USD / year                 |

Reviewer: Normally, budgetary approvals are "up to $X."  So, eg, we can approve "up to $15/year for email hosting through Google" (or whatever).

# Funding

## Scope and objectives

PHP-FIG is, as it often happens in Open Source communities, an organization composed by volunteers only; this doesn't exclude the fact that the organization itself may have some costs to sustain itself, for example maintaining tools such as a site and an email account. For such and similar reasons, we need a way to raise and govern money inside our organization.

This document defines the roles, rules and processes that must be followed inside the PHP-FIG in all matters regarding money, to guarantee fairness, transparency and impartiality. This document intentionally describes the only allowed processes and ways to raise, administer and spend money, so that any change to such processes or any variation in budget allocation requires a bylaw change vote. 

## Fiscal hosting

> For handling money and fiscal matters, the PHP-FIG will use [OpenCollective](https://opencollective.com/) as a fiscal host.

Reviewer 2: ```suggestion
For handling money and fiscal matters, the PHP-FIG will use [Open Collective](https://opencollective.com/) as a platform and [Open Source Collective](https://opencollective.com/opensource) as a fiscal host.
```

Author: I'll leave this suggestion pending until I'm able to investigate about the differences, but I'm inclined to approve it for now.

All three Secretaries, but only the Secretaries, will have full administrative access to PHP-FIG's OpenCollective account.  They do not have discretionary authority, however, except as stated explicitly in this bylaw.

> All monetary contributions will flow through OpenCollective, so that any tax considerations will be handled automatically, and every financial decision will be published and publicly available through OpenCollective's budgetary tools.

Reviewer 2: ```suggestion
All monetary contributions will flow through Open Collective, so that any tax considerations will be handled automatically, and every financial decision will be published and publicly available through Open Collective's budgetary tools.
```

The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119][].

## Fiscal hosting

For handling money and fiscal matters, the PHP-FIG MUST use [Open Collective](https://opencollective.com/) as a platform and [Open Source Collective](https://opencollective.com/opensource) as a fiscal host.
All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's OpenCollective account. They SHALL NOT use those permissions in a discretionary manner, except as stated explicitly in this bylaw.

All monetary contributions MUST flow through Open Collective, so that any tax considerations will be handled automatically, and every financial decision MUST be published and publicly available here and through Open Collective's budgetary tools.

> # Raising money

Reviewer 3: This probably should be H2 level, with double #?

PHP-FIG SHALL NOT solicit donations from any party, especially from private individuals. The only allowed funding sources MUST be listed below.

### Through OpenCollective

Direct contributions MUST NOT be accepted through OpenCollective; the collective home page MUST be set up in a way that:
 * all contribution sections and features are hidden and/or disabled, if possible;
 * explicit wording will be visible reminding visitors of the PHP-FIG stance in the matter.

### Through TideLift

Contributions will be accepted through [TideLift](https://tidelift.com/). All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's TideLift account, which MUST be configured to pay out to the Open Collective account. 

## Spending money

> Funds donated to the PHP-FIG MUST NOT be used for non-personnel operating expenses.  PHP-FIG SHALL NOT pay individual contributors to PHP-FIG standards or working groups.

Reviewer 3: This sounds like an improper double-negative? _MUST NOT .. non-personnel_

Reviewer 4: Should `non-personnel` here be `non-organizational`?

Author: I'm not a native speaker, but it sounded weird to me too. @Crell, WDYT? The wording was originally yours.

Author: Solved in d146b7b

All operating or capital expenses MUST be approved by an Approval Vote of the Core Committee.  The expense MUST include whether it is one time or recurring and at which frequency.  All expenses MUST be justified by their contribution to the mission of PHP-FIG.

Expenses approved by the Core Committee MUST be listed below, and reported in the Open Collective budget.

### Overfunding, and giving back to the community

PHP-FIG SHOULD maintain financial reserves to account for three years of the currently approved budget.
At least once a year, any surplus funds MUST be forwarded to the [PHP Foundation](https://opencollective.com/phpfoundation), as a way to further support the PHP ecosystem.

### Approved expenses

This section will contain a chronological list of approved budgeting decisions, either recurring or not.

> As an exception to Votes Bylaw, changes to this section only SHOULD NOT trigger a Bylaw Change Vote, but only an Approval Vote by the Core Committee, as explained in the "Spending Money" section.

Author: Done in 2dcc9c0

This document defines the roles, rules and processes that must be followed inside the PHP-FIG in all matters regarding money, to guarantee fairness, transparency and impartiality. This document intentionally describes the only allowed processes and ways to raise, administer and spend money, so that any change to such processes requires a bylaw change vote. 

The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119][].

## Fiscal hosting

For handling money and fiscal matters, the PHP-FIG MUST use [Open Collective](https://opencollective.com/) as a platform and [Open Source Collective](https://opencollective.com/opensource) as a fiscal host.

> All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's OpenCollective account. They SHALL NOT use those permissions in a discretionary manner, except as stated explicitly in this bylaw.

Reviewer: ```suggestion

All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's OpenCollective account. They SHALL NOT use those permissions in a discretionary manner, except as stated explicitly in this bylaw.
```

Consistent paragraph break style.

All funds MUST be collected into the Open Collective PHP-FIG account, so that any tax considerations will be handled automatically by the fiscal host, and every financial decision MUST be published and publicly available here and through Open Collective's budgetary tools.

## Raising money

PHP-FIG SHALL NOT solicit donations from any party, especially from private individuals; all exceptions to this rule MUST be listed below.

### Through OpenCollective

Direct contributions MUST NOT be accepted through Open Collective, to avoid private/personal donations; the collective home page MUST be set up in a way that:
 * all contribution sections and features are hidden and/or disabled, if possible;
 * explicit wording will be visible reminding visitors of the PHP-FIG stance in the matter (see "Scope and objectives").

### Through TideLift

> Contributions will be accepted through [TideLift](https://tidelift.com/). All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's TideLift account, which MUST be configured to pay out to the Open Collective account. 

Reviewer 5: ```suggestion
Contributions SHALL be accepted through [TideLift](https://tidelift.com/). All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's TideLift account, which MUST be configured to pay out to the Open Collective account. 
```

## Spending money

> Funds donated to the PHP-FIG MUST be used solely for non-personnel, operating expenses.  PHP-FIG SHALL NOT pay individual contributors to PHP-FIG standards or working groups.

Reviewer 5: Should we also mention secretaries and CC members here?

Author: Fixed in f4e4763

## Raising money

PHP-FIG SHALL NOT solicit donations from any party, especially from private individuals; all exceptions to this rule MUST be listed below.

### Through OpenCollective

Direct contributions MUST NOT be accepted through Open Collective, to avoid private/personal donations; the collective home page MUST be set up in a way that:
 * all contribution sections and features are hidden and/or disabled, if possible;
 * explicit wording will be visible reminding visitors of the PHP-FIG stance in the matter (see "Scope and objectives").

### Through TideLift

Contributions SHALL be accepted through [TideLift](https://tidelift.com/). All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's TideLift account, which MUST be configured to pay out to the Open Collective account. 

## Spending money

Funds donated to the PHP-FIG MUST be used solely for non-personnel, operating expenses.  PHP-FIG SHALL NOT pay individual contributors to PHP-FIG standards or other personnel, like Core Committee members, Secretaries, Project Representatives or working groups.

> All expenses MUST be approved by an Approval Vote of the Core Committee.  The expense MUST include whether it is one time or recurring and at which frequency.  All expenses MUST be justified by their contribution to the mission of PHP-FIG.

Reviewer 3: Proposed addition to accommodate for price changes:
```
Secretaries MAY request Implicit Approval from the Core Committee for changing the approved amount for previously accepted recurring expenses.
```

Author: Do we want to put a cap on this rule? Something like a percentage, 10% maybe?

I would rephrase that as:

> Secretaries MAY request an Implicit Approval to change the approved sum for a recurring expense, if it's a decrease or it does not exceed a 10% increase from the previously approved amount; higher increases or any other type of change MUST go through a regular Approval Vote.

Do you like this form? I also added an "any other type of change" so that i.e. a provider change has to be explicitly approved too.

Reviewer 3: I have no strong opinion on this. My intended wording would become:
> All expenses MUST be approved by an Approval Vote of the Core Committee.  The expense MUST include whether it is one time or recurring and at which frequency.  All expenses MUST be justified by their contribution to the mission of PHP-FIG.  Secretaries MAY request Implicit Approval from the Core Committee for changing the approved amount for previously accepted recurring expenses.

I think that is already clear enough that Implicit Approval is only applicable to price changes. If you'd want to limit the amount, you'd also have to specify a period. Alternatively, wording like this could be used:

> All expenses MUST be approved by an Approval Vote of the Core Committee.  The expense MUST include whether it is one time or recurring and at which frequency.  All expenses MUST be justified by their contribution to the mission of PHP-FIG.  In response to supplier price changes for previously accepted recurring expenses, Secretaries MAY request Implicit Approval from the Core Committee to raise the approved amount.

Reviewer: IMO the Implicit Approval raising should then be limited to recurring expenses.  For one-offs, if the CC approved $50, and the Secretaries then decide to spend $100, what was the point of the original CC vote?

Reviewer 6: @Crell A good argument, but Implicit Approval requires notification and a 7 day period for objection.  An example (though I'm not sure the FIG would ever hit it as It probably wouldn't be aligned with mission, but) would be some sort of customized merch order.  These are often priced per piece and the final order can be over/under those pieces by a certain percent for various reasons I've never taken the time to understand.  In that case, implicit approval would be reasonable.

Reviewer 6: 10% sounds reasonable to me, especially with US inflation hovering around 6-8%

Reviewer: Point of order: US inflation is now around 3%.  It has come back down rather dramatically in the past year. 😄

Author: 10% cap added! I did it only for recurring expenses, since I expect that to be solved by Secretaries' due diligence (and a bit of overhead) for one-shot expenses.

# Raising money

PHP-FIG SHALL NOT solicit donations from any party, especially from private individuals. The only allowed funding sources MUST be listed below.

### Through OpenCollective

Direct contributions MUST NOT be accepted through OpenCollective; the collective home page MUST be set up in a way that:
 * all contribution sections and features are hidden and/or disabled, if possible;
 * explicit wording will be visible reminding visitors of the PHP-FIG stance in the matter.

### Through TideLift

Contributions will be accepted through [TideLift](https://tidelift.com/). All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's TideLift account, which MUST be configured to pay out to the Open Collective account. 

## Spending money

Funds donated to the PHP-FIG MUST NOT be used for non-personnel operating expenses.  PHP-FIG SHALL NOT pay individual contributors to PHP-FIG standards or working groups.

> All operating or capital expenses MUST be approved by an Approval Vote of the Core Committee.  The expense MUST include whether it is one time or recurring and at which frequency.  All expenses MUST be justified by their contribution to the mission of PHP-FIG.

Reviewer 3: I the above line becomes `MUST be used for non-personnel operating expenses`, what would happen to capital expenses?

For handling money and fiscal matters, the PHP-FIG MUST use [Open Collective](https://opencollective.com/) as a platform and [Open Source Collective](https://opencollective.com/opensource) as a fiscal host.
All three Secretaries, but only the Secretaries, MUST have full administrative access to PHP-FIG's OpenCollective account. They SHALL NOT use those permissions in a discretionary manner, except as stated explicitly in this bylaw.

> All funds MUST be collected into the Open Collective PHP-FIG account, so that any tax considerations will be handled automatically by the fiscal host, and every financial decision MUST be published and publicly available here and through Open Collective's budgetary tools.

Reviewer: What does "here" mean in this context?  That's not clear.
